src/pudl/analysis/mcoe.py

```diff
@@ -233,6 +242,57 @@ def mcoe_generators_asset(
 ]
 
 
+@dataclass
+class McoeCheckSpec:
+    """A dataclass to hold the specification for a MCOE check."""
+
+    asset: str
+    max_null_fraction: float = 0.8
+    blocking: bool = True
+
+
+mcoe_asset_check_specs = [
+    McoeCheckSpec(
+        asset="out_eia__yearly_generators",
+        max_null_fraction=0.9,
+    ),
+    McoeCheckSpec(
+        asset="out_eia__monthly_generators",
+        max_null_fraction=0.9,
+    ),
+    McoeCheckSpec(
+        asset="_out_eia__yearly_derived_generator_attributes",
+        max_null_fraction=0.8,
+    ),
+    McoeCheckSpec(
+        asset="_out_eia__monthly_derived_generator_attributes",
+        max_null_fraction=0.8,
+    ),
+]
+
+
+def mcoe_asset_check_factory(spec: McoeCheckSpec) -> AssetChecksDefinition:
+    """Turn a MCOE check spec into an AssetChecksDefinition."""
+
+    @asset_check(asset=spec.asset, blocking=spec.blocking)
+    def excessively_null_rows(df: pd.DataFrame) -> AssetCheckResult:
+        """Check that the MCOE dataframe has no excessively null rows."""
+        excessively_null_rows = (
+            df.isna().sum(axis="columns") / len(df.columns) > spec.max_null_fraction
+        )
```

Comment:
In poking around at this, I found that we're actually using `pv.no_null_rows` as a validation step in some big merges in e.g. the Plant Parts EIA module, as well as below in this one).  At the moment this core checking code is duplicated, since the other validation function is set up to be usable with `df.pipe()` which isn't really compatible with the use case here. But maybe the core logic can be factored out and serve both purposes without duplication.

Reply:
why isn't using `pv.no_null_rows` compatible here? we use it below with `pipe` but that doesn't mean it needs to be used that way.

Reply:
Because it will `raise` an exception if it finds null rows, rather than allowing us to construct an `AssetCheckResult` to return.

But actually duh, I could just try/except and catch that exception.